packages/amplify-graphql-conversation-transformer/src/resolvers/write-message-to-table-resolver.ts

```diff
@@ -1,63 +1,23 @@
 import { MappingTemplate } from '@aws-amplify/graphql-transformer-core';
 import { MappingTemplateProvider } from '@aws-amplify/graphql-transformer-interfaces';
-import { dedent } from 'ts-dedent';
+import fs from 'fs';
+import path from 'path';
 
 /**
  * Creates a mapping template for writing a message to a table in a conversation.
  *
- * @param {string} fieldName - The name of the field to write to the table.
  * @returns {MappingTemplateProvider} An object containing request and response MappingTemplateProviders.
  */
 export const writeMessageToTableMappingTemplate = (fieldName: string): MappingTemplateProvider => {
-  const req = createWriteMessageToTableRequestFunction(fieldName);
-  const res = createWriteMessageToTableResponseFunction();
-  return MappingTemplate.inlineTemplateFromString(dedent(req + '\n' + res));
-};
-
-/**
- * Creates the request function for writing a message to a table in a conversation.
- *
- * @param {string} fieldName - The name of the field to write to the table.
- * @returns {MappingTemplateProvider} A MappingTemplateProvider for the request function.
- */
-const createWriteMessageToTableRequestFunction = (fieldName: string): string => {
-  const requestFunctionString = `
-    import { util } from '@aws-appsync/utils'
-    import * as ddb from '@aws-appsync/utils/dynamodb'
-
-    export function request(ctx) {
-      const args = ctx.stash.transformedArgs ?? ctx.args;
-      const defaultValues = ctx.stash.defaultValues ?? {};
-      const message = {
-          __typename: 'ConversationMessage${fieldName}',
-          role: 'user',
-          ...args,
-          ...defaultValues,
-      };
-      const id = ctx.stash.defaultValues.id;
-
-      return ddb.put({ key: { id }, item: message });
-    }
-    `;
-
-  return requestFunctionString;
-};
-
-/**
- * Creates the response function for writing a message to a table in a conversation.
- *
- * @returns {MappingTemplateProvider} A MappingTemplateProvider for the response function.
- */
-const createWriteMessageToTableResponseFunction = (): string => {
-  const responseFunctionString = `
-    export function response(ctx) {
-      if (ctx.error) {
-        util.error(ctx.error.message, ctx.error.type);
-      } else {
-        return ctx.result;
-      }
-    }
-    `;
+  const substitutions = {
+    CONVERSATION_MESSAGE_TYPE_NAME: `ConversationMessage${fieldName}`,
+  };
+  let resolver = fs.readFileSync(path.join(__dirname, 'write-message-to-table-resolver-fn.js'), 'utf8');
+  Object.entries(substitutions).forEach(([key, value]) => {
```

Comment:
nit for future PR: recommend centralizing substitution logic in a utility somewhere. That util can take care of the file renaming logic as well to help enforce the convention.

Reply:
💯  that's the plan. I felt this PR was already too large as is and want to do this in a separate change.